src/index.js

```diff
@@ -1,13 +1,77 @@
 // @flow
 
 import type {
+  NavigationAction,
+  NavigationContainer,
   NavigationEventCallback,
   NavigationState,
 } from 'react-navigation';
-import type { Middleware } from 'redux';
+import type { Middleware, Reducer } from 'redux';
 
+import { NavigationActions } from "react-navigation";
 import invariant from 'invariant';
 
+type Navigator = NavigationContainer<*, *, *>;
+type ReducerState = ?NavigationState;
+type ScreenTrackingConfig<State> = {
+  navStateSelector: (state: State) => NavigationState,
+  onScreenChange: (?string, ?string) => any
+};
+
+function getCurrentRouteName(navigationState: ReducerState) {
```

Comment:
@Ashoat I can't find the correct type for this, sorry.

I can't use this one right now

```
[flow] Cannot call `getCurrentRouteName` with `route` bound to `navigationState` because property `index` is missing in `NavigationLeafRoute` [1] but exists in `NavigationState` [2]. (References: [1] [2])
[flow] Cannot call `getCurrentRouteName` with `route` bound to `navigationState` because property `routes` is missing in `NavigationLeafRoute` [1] but exists in `NavigationState` [2]. (References: [1] [2])
```